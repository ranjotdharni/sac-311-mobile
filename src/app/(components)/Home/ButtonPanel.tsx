import { FontAwesome } from "@expo/vector-icons";
import { View, StyleSheet, Dimensions, TouchableOpacity, Linking, Text, Image} from "react-native";
import { global, shadowUniversal } from "../../../customs";
import { globalFont } from "../../../customs";
import { fontGetter } from "../../../customs";
import CustomText  from "../CustomText";
import { router } from "expo-router";
import { globalColorTheme } from '../../../customs';

// Extending the component's props to include 'style'
type ButtonProps = {
    testID: string;
    text: string;
    logo: string;
    path: string;
    style?: object | object[];
};

function Button({testID, text, logo, path, style }: ButtonProps) {
    return (
        <TouchableOpacity testID={testID} onPress={() => { (router.push as any)({ pathname: path }) }} style={[{width: '30%', aspectRatio: 1 / 1, backgroundColor: globalColorTheme.backgroundColor, borderRadius: 10, justifyContent: 'center', alignItems: 'center',}, style]}>
            <FontAwesome name={logo as any} size={Dimensions.get('screen').width * 0.15} color={globalColorTheme.blue} />
            <CustomText nol={0} font={fontGetter()} style={{color: globalColorTheme.blue, fontSize: Dimensions.get('screen').width * 0.045, fontFamily: globalFont.chosenFont}} text={text} />
        </TouchableOpacity>
    );
}

export default function ButtonPanel() {
    return (
        <View style={{backgroundColor: globalColorTheme.backgroundColor, width: Dimensions.get('screen').width, height: Dimensions.get('screen').height * 0.22}}>
            <CustomText nol={0} font={fontGetter()} text='Are You A...' style={{color: globalColorTheme.blue, alignSelf: 'flex-start', fontSize: 25, fontFamily: globalFont.chosenFont, fontWeight: 'bold', paddingTop: '2%', paddingLeft: '3%',}} />
            
            <View style={{width: '100%', height: '85%', display: 'flex', flexDirection: 'row', justifyContent: 'space-evenly', alignItems: 'center', position: 'relative', top: '-2.5%',}}>
                <Button testID="homeBtn" text='Resident' logo='home' path="/(tabs)/ResidentTab" style={{shadowColor: globalColorTheme.color, shadowOffset: {width: -2, height: 4}, shadowOpacity: 0.33, shadowRadius: 10, elevation: 5}}/>
                <Button testID="visBtn" text='Visitor' logo='location-arrow' path="/(tabs)/VisitorTab" style={{shadowColor: globalColorTheme.color, shadowOffset: {width: -2, height: 4}, shadowOpacity: 0.33, shadowRadius: 10, elevation: 5}}/>
                <Button testID="bizBtn" text='Business' logo='suitcase'  path="/(tabs)/BusinessTab" style={{shadowColor: globalColorTheme.color, shadowOffset: {width: -2, height: 4}, shadowOpacity: 0.33, shadowRadius: 10, elevation: 5}}/>
            </View>
        </View>
    )
}


