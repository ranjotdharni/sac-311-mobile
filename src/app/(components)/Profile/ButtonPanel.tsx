import React from 'react';
import { View, StyleSheet, Dimensions, TouchableOpacity, Text } from 'react-native';
import CustomText from '../CustomText';
import { fontGetter, global, salesforceDevelopmentSignature } from '../../../customs';
import { fontSetter } from '../../../customs';
import { globalFont } from '../../../customs';
import { Link, usePathname, router } from "expo-router";

interface ButtonProps {
  text: string;
  buttonColor: string;
  textColor: string;
  onPress: () => void; // Add onPress prop
}

function Button({ text, buttonColor, textColor, onPress }: ButtonProps) {
  return (
    <View style={{ ...styles.ButtonWrapper, backgroundColor: buttonColor }}>
      <TouchableOpacity style={styles.TouchableWrapper} onPress={onPress}>
        <CustomText nol={0} font={fontGetter()} style={{ ...styles.ButtonText, color: textColor }} text={text} />
      </TouchableOpacity>
    </View>
  );
}

function ButtonDumb({ text, buttonColor, textColor, onPress }: ButtonProps) {
  return (
    <View style={{ ...styles.ButtonWrapper, backgroundColor: buttonColor }}>
      <TouchableOpacity style={styles.TouchableWrapper} onPress={onPress}>
        <Text style={{ ...styles.ButtonText, color: textColor }}>{text}</Text>
      </TouchableOpacity>
    </View>
  );
}

export default function ButtonPanel({ onPressLoginSignup }: { onPressLoginSignup: () => void },) {
  return (
    <View style={styles.ButtonPanelWrapper}>
      <Button text="Login or Signup" buttonColor="white" textColor="grey" onPress={onPressLoginSignup} />
      <ButtonDumb
        text="Submit New Request"
        buttonColor={global.baseBlue100}
        textColor="white"
        onPress={() => {}}
      >
        <Link
          to={{
            pathname: '/request/Type',
            params: {
              Subject: 'salesforceDevelopmentSignature',
              Service_Type__c: '', // CategoryLevel1
              Sub_Service_Type__c: '', // CategoryLevel2
              Council_District__c: '', // CouncilDistrictNumber
              GIS_Street_Address__c: '', // CrossStreet
              GIS_Zip_Code__c: '', // ZIP
              Address__c: '', // Address
              GIS_System_Info__c: '311 Phone', // "<Data_Source>  <SourceLevel1>"
              GIS_Neighborhood_Name__c: '', // Neighborhood
              description: '',
              Address_Geolocation__Latitude__s: 0,
              Address_Geolocation__Longitude__s: 0,
              returnRoute: usePathname().replace('/', ''),
            },
          }}
          style={styles.NewRequestLink}
        >
          <Text style={styles.requestText}>Submit New Request</Text>
        </Link>
      </ButtonDumb>
    </View>
  );
}

const styles = StyleSheet.create({
    ButtonPanelWrapper: {
        width: Dimensions.get('screen').width,
        height: Dimensions.get('screen').height * 0.22,
        backgroundColor: 'transparent', // Change this line
        position: 'absolute',
        bottom: 110,
        alignItems: 'center', // added this line to center the buttons
    },
    
    ButtonWrapper: {
        width: '92.5%',
        height: '20%',
        marginVertical: '3%',
        borderRadius: 10,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: 'transparent',
        shadowColor: "#000", 
        shadowOffset: { 
            width: -2,
            height: 2,
        },
        shadowOpacity: 0.25, 
        shadowRadius: 3.84, 
        elevation: 7.5, 
    },
    
    TouchableWrapper: {
        width: '100%',
        height: '100%',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#0000', // Add this line
    },     

    ButtonText: {
        fontSize: Dimensions.get('screen').width * 0.045,
        fontFamily: globalFont.chosenFont
    },
})