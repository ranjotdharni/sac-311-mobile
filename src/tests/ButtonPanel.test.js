import renderer from "react-test-renderer";
import ButtonPanel from '../app/(components)/Home/ButtonPanel'; 

describe('ButtonPanel', () => {
  it('make sure the panel renders', () => {
    //const { getByTestId } = render(<ButtonPanel />);
    const tree = renderer.create(<ButtonPanel />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("has 1 children", () => {
    const tree = renderer.create(<ButtonPanel />).toJSON();
    expect(tree.children.length).toBe(1);
  });
});
