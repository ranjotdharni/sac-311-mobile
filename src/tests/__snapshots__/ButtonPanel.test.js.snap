// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPanel make sure the panel renders 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
      "height": 293.48,
      "width": 750,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "height": "85%",
        "justifyContent": "space-evenly",
        "position": "relative",
        "top": "-2.5%",
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "aspectRatio": 1,
          "backgroundColor": "#ffffff",
          "borderRadius": 10,
          "elevation": 5,
          "justifyContent": "center",
          "opacity": 1,
          "shadowColor": "#000000",
          "shadowOffset": {
            "height": 4,
            "width": -2,
          },
          "shadowOpacity": 0.33,
          "shadowRadius": 10,
          "width": "30%",
        }
      }
      testID="homeBtn"
    >
      <Text />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "aspectRatio": 1,
          "backgroundColor": "#ffffff",
          "borderRadius": 10,
          "elevation": 5,
          "justifyContent": "center",
          "opacity": 1,
          "shadowColor": "#000000",
          "shadowOffset": {
            "height": 4,
            "width": -2,
          },
          "shadowOpacity": 0.33,
          "shadowRadius": 10,
          "width": "30%",
        }
      }
      testID="visBtn"
    >
      <Text />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "aspectRatio": 1,
          "backgroundColor": "#ffffff",
          "borderRadius": 10,
          "elevation": 5,
          "justifyContent": "center",
          "opacity": 1,
          "shadowColor": "#000000",
          "shadowOffset": {
            "height": 4,
            "width": -2,
          },
          "shadowOpacity": 0.33,
          "shadowRadius": 10,
          "width": "30%",
        }
      }
      testID="bizBtn"
    >
      <Text />
    </View>
  </View>
</View>
`;
